sdk/program/src/message/versions/mod.rs

```diff
@@ -198,7 +198,16 @@ impl<'de> Deserialize<'de> for MessagePrefix {
                 formatter.write_str("message prefix byte")
             }
 
-            fn visit_u8<E>(self, byte: u8) -> Result<MessagePrefix, E> {
+            // Serde's integer visitors bubble up to u64 so check the prefix
+            // with this function instead of visit_u8. This approach is
+            // necessary because serde_json directly calls visit_u64 for
+            // unsigned integers.
+            fn visit_u64<E: de::Error>(self, value: u64) -> Result<MessagePrefix, E> {
```

Comment:
does this not break bincode?

Reply:
No because integer visitors bubble up to u64. By not implementing `visit_u8`, we allow serde to bubble up `visit_u8` -> `visit_u16` -> `visit_u32` -> `visit_u64` and handle it there instead.

Reply:
We have tests for bincode already and they didn't break. The problem was that before we never tested `serde_json` but now both `serde_json` and `bincode` are under test.

Reply:
ah. that's unintuitive. i figured it'd consume eight bytes